### PR TITLE
Further refactor of menu managment

### PR DESCRIPTION
**Pull request by ShFil119**
It also fixes weird behavior which can happen it debug mode on master. (freeze/crash) Caused by #667.

### CODE REVIEW

File: rwgame/MenuSystem.hpp [hunk 1]
```diff
@@ -71,18 +71,14 @@ class Menu {
...
 
-    Menu(std::vector<MenuEntry> initial, int font = MenuDefaults::kFont,
+    Menu(std::initializer_list<MenuEntry>&& initial,
+         const glm::vec2& offset = glm::vec2(), int font = MenuDefaults::kFont,
          float size = 30.f)
-        : activeEntry(-1), font(font), size(size), entries(std::move(initial)) {
-    }
-
-    /**
-     * @brief creates a menu from the given menu items
```
Comment (dracc) [inline rwgame/MenuSystem.hpp]: Please consider adding some kind of comment back.

File: tests/test_Menu.cpp [hunk 2]
```diff
@@ -27,9 +27,7 @@ BOOST_AUTO_TEST_CASE(menu_test_click) {
 
 BOOST_AUTO_TEST_CASE(menu_test_click_offset) {
     bool clickered = false;
```
Comment (madebr) [inline tests/test_Menu.cpp]: nit: clicked?

File: rwgame/MenuSystem.hpp (hunk 1 above)
Comment (ShFil119) [reply rwgame/MenuSystem.hpp]: Earlier to was wrapper to create ptr of `Menu` class, for optional container, plain constructor is enough.

**Comment by ShFil119** [comment]
@madebr done. ;)

**Pull request merged**